ra/ra.go

```diff
@@ -1222,6 +1222,19 @@ func (ra *RegistrationAuthorityImpl) enforceNameCounts(
 }
 
 func (ra *RegistrationAuthorityImpl) checkCertificatesPerNameLimit(ctx context.Context, names []string, limit ratelimit.RateLimitPolicy, regID int64) error {
+	if features.Enabled(features.CheckRenewalFirst) {
```

Comment:
minor nit: rather than duplicate the code in the `features.Enabled(features.CheckRenewalFirst)` branch and the `!features.Enabled(features.CheckRenewalFirst)` branch could you lift the code into a function that both call?

Reply:
I think that function would not actually simplify the code. It would have to return two values: a boolean `exists` and an error. The call sites would have to have two `if` statements: If the error is non-nil, return the error, and if `exists` is true, return `nil`. That's the exact same complexity as the current code. The only thing we'd save is the format string for the error and the stats increment (which I think we should probably remove anyhow since we don't use it and it's less meaningful now with the reordered checks).

Reply:
Ok :+1: